crates/node/src/catchup.rs

```diff
@@ -263,6 +280,29 @@ impl Node {
         message: types::CatchupStreamMessage,
     ) -> eyre::Result<Option<calimero_primitives::context::Context>> {
         match message {
+            types::CatchupStreamMessage::Request(request) => {
+                // Handle the initial catchup request
+                info!(
+                    ?request,
+                    "Received catchup request for context {}", request.context_id
+                );
+
+                // TODO: Consider storing the public key in the future? E.g.
+                // self.store_peer_public_key(chosen_peer, request.public_key)?;
+
+                // If we don't have a context yet, create one
+                if context.is_none() {
+                    context = Some(calimero_primitives::context::Context {
+                        id: request.context_id,
+                        application_id: request.application_id.ok_or_else(|| {
+                            eyre::eyre!("Application ID missing in catchup request")
+                        })?,
+                        last_transaction_hash: request.last_executed_transaction_hash,
+                    });
+                }
+
+                // Note: Might need to prepare and send an initial response?
+            }
```

Comment:
we can skip doing this if the initial catchup request includes all the identities on our node